### PR TITLE
Fix spacings around `!`and `?`

### PR DESCRIPTION
**Pull request by kimdv**
Fixes some spacing regressions found in #1217

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — COMMENTED** [review]
This makes me wonder whether the better default would be to mark `exclamationMark` and `postfixQuestionMark` as not having a trailing space and selectively enabling it in cases where we want a trailing space.

I’m thinking that in the most cases the exclamation and question mark will be appended to an identifier and since identifiers don’t have a trailing space, `!` and `?` also shouldn’t have a trailing space. I might be wrong though.

**Comment by kimdv** [comment]
I can see what happens 😬

**Comment by kimdv** [comment]
@swift-ci please test

File: CodeGeneration/Sources/generate-swiftsyntax/templates/basicformat/BasicFormatFile.swift [hunk 1]
```diff
@@ -157,18 +157,16 @@ let basicFormatFile = SourceFile {
...
-             (.postfixQuestionMark, .rightAngle), // Ensures there is not space in `ContiguousArray<RawSyntax?>`
-             (.postfixQuestionMark, .rightParen), // Ensures there is not space in `myOptionalClosure?()`
              (.tryKeyword, .exclamationMark), // Ensures there is not space in `try!`
              (.tryKeyword, .postfixQuestionMark): // Ensures there is not space in `try?`
           return false
         case (.spacedBinaryOperator, .comma):
           return false
+        case (.exclamationMark, .identifier), // Ensures there is not space in `try! foo`
+             (.postfixQuestionMark, .identifier): // Ensures there is not space in `try? foo`
+          return true
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/basicformat/BasicFormatFile.swift:169]: Wouldn’t it make more sense to do 
```swift
switch (token.kind, token.parent?.kind) {
  case (.colon, .dictionaryExpr): // Covers the case at the top of this function
    return false
  case (.exclamationMark, .tryExpr), // Ensures there is a space in `try! foo`
       (.postfixQuestionMark, .tryExpr): // Ensures there is a space in `try? foo`
    return true
}
```

That should also cover cases like `try! (foo())` or `try! !functionThatThrows()`.

**Review by ahoppen — COMMENTED** [review]
I like this much better.

**Comment by kimdv** [comment]
@swift-ci please test

File: CodeGeneration/Sources/generate-swiftsyntax/templates/basicformat/BasicFormatFile.swift (hunk 1 above)
Comment (ahoppen) [reply CodeGeneration/Sources/generate-swiftsyntax/templates/basicformat/BasicFormatFile.swift]: Yes, exactly. Looks good!

**Review by ahoppen — APPROVED** [review]

**Pull request merged**